vendors/st/boards/stm32l475_discovery/ports/secure_sockets/iot_secure_sockets.c

```diff
@@ -564,6 +704,9 @@ Socket_t SOCKETS_Socket( int32_t lDomain,
         xSockets[ ulSocketNumber ].pvTLSContext = NULL;
         xSockets[ ulSocketNumber ].pcServerCertificate = NULL;
         xSockets[ ulSocketNumber ].ulServerCertificateLength = 0;
+        #if ( configPLATFORM_SOCKET_UDP_SUPPORT == 1 )
+            xSockets[ ulSocketNumber ].usLocalPort = -1;
```

Comment:
usLocalPort is unsigned int,  this will be a large positive number and also could be a valid port address.

Reply:
Good catch, this was set to 0 originally but test binds to port 0 and doing so will result in binding to the same port. But now I have a solution in mind that should disregard the default value.